### ISSUE TITLE
Error prone not giving any warnings (Java 8)

### ISSUE THREAD

**Issue by ankit-1517** [google/error-prone#1755]
### Description of the problem:

I take the code given for `InfiniteRecursion` [here](https://errorprone.info/bugpattern/InfiniteRecursion), and try to see what kind of warning/error is reported by the error prone tool. Much to my surprise, I don't get any output (i.e., no "problem" in the code). And it's not just for `InfiniteRecursion`, but for many many others -- `ArrayEquals`, `ArrayToString`, etc. to name a few. And it's not the case that none of the bug patterns work either. For instance, I tried `CollectionIncompatibleType` from [here](https://errorprone.info/bugpattern/CollectionIncompatibleType), and I can reproduce the warning discussed. Roughly, error prone doesn't give the said warning 9/10 times (it's not practical for me to do it for all bug patterns, so i tried only for a small subset).

### Bugs: what's the simplest, easiest way to reproduce this bug?

```
public class ep_errors{
  public static int oops() {
    return oops();
  }

  public static void main(String[] args) {
    oops();
  }
}
```

### What version of Error Prone are you using?

2.4.0
OpenJDK and OpenJRE: 1.8.0_252

###  Have you found anything relevant by searching the web?

No

**Comment by ankit-1517** [comment]
I didn't realise I was using `DisableAllChecks` flag. Works fine now.
